mteb/evaluation/evaluators/Image/ImageTextPairClassificationEvaluator.py

```diff
@@ -54,47 +95,74 @@ def __call__(
         if "batch_size" not in encode_kwargs:
             encode_kwargs["batch_size"] = 64
 
-        num_samples = len(self.images)
-        num_images_per_sample = len(self.images[0])
-        num_texts_per_sample = len(self.texts[0])
-
-        images = list(itertools.chain.from_iterable(self.images))
-        texts = list(itertools.chain.from_iterable(self.texts))
-
-        image_embeddings = F.normalize(
-            model.get_image_embeddings(images, batch_size=encode_kwargs["batch_size"]),
-            dim=-1,
-        ).view(num_samples, num_images_per_sample, -1)
-        text_embeddings = F.normalize(
-            model.get_text_embeddings(texts, batch_size=encode_kwargs["batch_size"]),
-            dim=-1,
-        ).view(num_samples, num_texts_per_sample, -1)
+        data_loader = DataLoader(
```

Comment:
I feel like the batching should be a model concern not a concern of the task.

We do this in main using `encode_kwargs`.

Reply:
I would love to hear what others think though

Reply:
I agree

Reply:
The current pipeline is quite aligned with main I think? If we do `results = evaluation.run(model, encode_kwargs={"batch_size": 1024}, output_folder="results"` for this task, it will be able to adjust as well (the batch size has been used in the construction of `ImageTextDataset` here). 

What is confusing might be that for other mieb tasks, we can do something like `model.get_image_embeddings(images, batch_size=encode_kwargs["batch_size"])` but for this task there is a loop over dataloader to get the images and texts out seperately first (image and text are currently both in the same Dataset Object) so it might look like the batch size is not used in the encoding process. I will aim to make it look better in the next commit.